File: python/tests/exceptions.cc
#include "LaDaConfig.h"

#include <boost/python/module.hpp>
#include <boost/python/def.hpp>

#include <root_exceptions.h>
#include "../exceptions.h"

namespace bp = boost::python;

void nothrow () {};
void dothrow_nomessage() { BOOST_THROW_EXCEPTION(::LaDa::error::LADA_TYPE()); }
void dothrow_message()
{
  std::string message = "This is a message.";
  BOOST_THROW_EXCEPTION(::LaDa::error::LADA_TYPE() << LaDa::error::string(message)); 
}


BOOST_PYTHON_MODULE(LADA_MODULE)
{
  bp::def("nothrow", &nothrow);
  bp::def("dothrow_nomessage", &dothrow_nomessage);
  bp::def("dothrow_message", &dothrow_message);
}

File: python/tests/exceptions.py
import sys
sys.path = ["@CMAKE_CURRENT_BINARY_DIR@"] + sys.path # "@PROJECT_BINARY_DIR@/testdir/", 
print sys.path
from lada import error
import exception_@TYPE@

exception_@TYPE@.nothrow()
try: exception_@TYPE@.dothrow_nomessage()
except error.@TYPE@: pass
try: exception_@TYPE@.dothrow_message()
except error.@TYPE@ as e: 
  assert str(e).find("This is a message.") != -1
try: exception_@TYPE@.dopythrow_message()
except error.@TYPE@ as e: 
  assert str(e).find("This is another message.") != -1

try: raise error.@TYPE@("Whatever")
except error.@TYPE@ as e: 
  assert str(e).find("Whatever") != -1
